### PR TITLE
Fix infinite recursion bug in error_destroy

### LINKED ISSUE #1249: StackOverflowException: new Error()
Description
===========
In at least one situation, Error(IntPtr error) ctor causes a StackOverflowException. For my specific case, it was because the TransactionTimeout was set to a value that was too large.

Client Version: 1.4
OS: Windows 10

Stack Trace:
```
Confluent.Kafka.dll!Confluent.Kafka.Impl.Librdkafka.error_destroy(System.IntPtr error) Line 1301
Confluent.Kafka.dll!Confluent.Kafka.Impl.Librdkafka.error_destroy(System.IntPtr error) Line 1301
Confluent.Kafka.dll!Confluent.Kafka.Impl.Librdkafka.error_destroy(System.IntPtr error) Line 1301
Confluent.Kafka.dll!Confluent.Kafka.Impl.Librdkafka.error_destroy(System.IntPtr error) Line 1301
Confluent.Kafka.dll!Confluent.Kafka.Impl.Librdkafka.error_destroy(System.IntPtr error) Line 1301
Confluent.Kafka.dll!Confluent.Kafka.Error.Error(System.IntPtr error) Line 67
Confluent.Kafka.dll!Confluent.Kafka.Impl.SafeKafkaHandle.InitTransactions(int millisecondsTimeout) Line 495
Confluent.Kafka.dll!.Confluent.kafka.Producer.InitTransactions()
```

How to reproduce
================
```
using Confluent.Kafka;
using System;

namespace SimpleProject
{
    class Program
    {
        static void Main(string[] args)
        {
            const int TOO_LARGE_VALUE = 1000 * 60 * 60 * 5;
            var clientConfig = new ClientConfig()
            {
                BootstrapServers = "localhost:9092",
            };

            var producerConfig = new ProducerConfig(clientConfig)
            {
                TransactionalId = "test-trx-id",
                TransactionTimeoutMs = TOO_LARGE_VALUE
            };
            using (var producer = new ProducerBuilder<string, string>(producerConfig).Build())
            {
                producer.InitTransactions(TimeSpan.FromHours(5));
            }
        }
    }
}

```

Checklist
=========

Please provide the following information:
 - [x] A complete (i.e. we can run it), minimal program demonstrating the problem. No need to supply a project file.
 - [x] Confluent.Kafka nuget version.
 - [ ] Apache Kafka version.
 - [x] Client configuration.
 - [x] Operating system.
 - [ ] Provide logs (with "debug" : "..." as necessary in configuration).
 - [ ] Provide broker log excerpts.
 - [ ] Critical issue.

### PR DESCRIPTION
**Pull request by mhowlett**
resolves #1249

### CODE REVIEW

File: src/Confluent.Kafka/Impl/LibRdKafka.cs [hunk 1]
```diff
@@ -1298,6 +1298,6 @@ internal static bool error_txn_requires_abort(IntPtr error)
 
         private static Action<IntPtr> _error_destroy;
         internal static void error_destroy(IntPtr error)
-            => error_destroy(error);
+            => _error_destroy(error);
```
Comment (edenhill) [inline src/Confluent.Kafka/Impl/LibRdKafka.cs:1301]: One byte fix :+1:

**Comment by mhowlett** [comment]
i'm not going to get to that now, it'll just delay 1.4.4

**Comment by mhowlett** [comment]
to clarify: 

1. in order to test for this, I need to arrange for an error scenario. I don't immediately have any great ideas on how to do that easily (there may be though that i'm missing).
2. i don't think there's a whole lot of value in regression testing this. there's not much to it at the .net level, and you'd need to go out of your way to screw it up.

sure, having a test is better given unlimited time, but I feel focusing on this is a misallocation of resources - we don't have unlimited time and we're better off spending it on things that have higher payoff (whether that's doing things that improve quality elsewhere or new features).

by contrast, this one line change has massive ROI. ROI is where it's at!

side note: I actually don't think we have a problem with misallocation of resources in our work on the clients in general in this way. Its kind of a sore point from the past. I've seen projects completely screwed up by people doing masses of low value testing (generally because it's easy), whilst not doing basic testing where it's going to payoff.

**Comment by mhowlett** [comment]
ok, added a test, it is in fact easy.

File: test/Confluent.Kafka.IntegrationTests/Tests/Transactions_Error.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+#pragma warning disable xUnit1026
+
+using System;
+using Xunit;
+
+
```
Comment (edenhill) [inline test/Confluent.Kafka.IntegrationTests/Tests/Transactions_Error.cs:22]: There's also a `rd_kafka_error_new()` that allows you to construct an error object.

**Pull request merged**